tests/integration/test_vulnerability_detector/test_scan_results/test_scan_different_cves.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import pytest
+import json
+from time import sleep
+
+import wazuh_testing.vulnerability_detector as vd
+from wazuh_testing.tools import LOG_FILE_PATH
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools.services import control_service
+
+# Marks
+pytestmark = pytest.mark.tier(level=1)
+
+
+# Variables
+current_test_path = os.path.dirname(os.path.realpath(__file__))
+test_data_path = os.path.join(current_test_path, 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_different_cves.yaml')
+vulnerabilities_data_path = os.path.join(test_data_path, vd.VULNERABILITIES_DIFFERENT_CVE)
+
+
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+SCAN_TIMEOUT = 40
+
+
+# Set configuration
+parameters = [{'NVD_JSON_PATH': os.path.join(test_data_path, vd.CUSTOM_NVD_FEED)}]
+metadata = [{'nvd_json_path': os.path.join(test_data_path, vd.CUSTOM_NVD_FEED)}]
+ids = ['test_different_cves']
+
+
+# Read JSON data templates
+with open(vulnerabilities_data_path, 'r') as f:
+    vulnerabilities = json.loads(f.read())
+
+vulnerabilities_nvd = vulnerabilities['vulnerabilities_nvd']
+vulnerabilities_provider = vulnerabilities['vulnerabilities_provider']
+
+rpm_expected_alerts_nvd = sum(1 for i in vulnerabilities_nvd if i['vulnerable'] == "rpm" or i['vulnerable'] == "all")
```

Comment:
Why not use the the length of the list?

Reply:
We don't expect all the packages to generate vulnerabilities. Every package is in both list, NVD and provider ones, and some of them will generate alerts and others don't. It is also diferent for rpm and deb.

Reply:
Thanks! https://github.com/wazuh/wazuh-qa/pull/801/commits/78d2d4ae31011023617d8d6fc62f352ac77f86cc

Reply:
```suggestion
rpm_expected_alerts_nvd = len(list(filter(lambda x: x['vulnerable'] in ['all', 'rpm'], vulnerabilities_nvd)))
```